Kudu.Services/Infrastructure/UriHelper.cs

```diff
@@ -1,9 +1,29 @@
 ﻿using System;
+using System.Collections.Generic;
+using System.Globalization;
+using System.Linq;
+using System.Net.Http;
 
 namespace Kudu.Services.Infrastructure
 {
     public static class UriHelper
     {
+        public static Uri GetBaseUri(HttpRequestMessage request)
+        {
+            if (request == null)
+            {
+                throw new ArgumentNullException("request");
```

Comment:
We don't typically do argument checks for things we only call ourselves. This is not a framework.